### ISSUE TITLE
Make Lutron scenes accessible from HomeKit?

### ISSUE THREAD

**Issue by Launchbury** [rnilssoncx/homebridge-pico#44]
I very much appreciate the homebridge-pico. It provides significant critical functionality. Thank you!

A major new piece of desirable functionality would be to enable triggering Lutron scenes from HomeKit. These scenes have the significant advantage that Lutron has worked out a very elegant synchronized method for multiple lights/devices all responding together (through programming the switches ahead of time). HomeKit will never be able to replicate this through its independent device communication model.

To enable HomeKit to trigger Lutron scenes requires some new infrastructure in the homebridge-pico setup to send info in the other direction, i.e. receive requests from HomeKit and send them to the Caseta hub. Additionally, the scenes need to show up in HomeKit – I suspect they will just have to appear as new devices rather than HomeKit scenes, but that should work pretty well. On the Lutron hub, I think the scenes are DEVICE 0, with each of the scenes having a separate trigger number, so I think that communication will be fairly straightforward to establish.

I'd be happy to help with programming this, but I can't get started as I don't know enough to set up the basic communication infrastructure from HomeKit to Lutron. Once that is in place I could help with the details of the messages to the hub.

Thanks for considering this!

**Comment by rnilssoncx** [comment]
Do you see anything on the bus (if you have full logging enabled) when a scene is fired?

This may not be possible with the way I'm integrated.

**Comment by Launchbury** [comment]
No, there's nothing on the bus even with full logging. But if I do a direct telnet connection I can invoke a scene using a command to the Device 0.
